server/routes/users.ts

```diff
@@ -88,11 +88,29 @@ router.post('/signUp', async (req, res) => {
     return res.json({ message: 'Signed up.' });
 });
 
-// TODO:
 // PATCH /settings
 router.patch('/settings', async (req, res) => {
-    // TODO: change pfp??
-    // TODO: change user's name??
+    if(req.session.user) { 
+        // TODO make sure a user cannot edit another user's info (req.params.id != req.session.user._id)
+
+        if(!req.body.email || !checkEmail(req.body.email)) throw new Error("[routes/users updateUser] email is invalid.");
+        if(!req.body.name || !checkNonEmptyString(req.body.name)) throw new Error("[routes/users updateUser] name is invalid.");
+        if(!req.body.pfp || !checkNonEmptyString(req.body.pfp) || !checkURL(req.body.pfp)) throw new Error("[routes/users updateUser] pfp link is invalid.");
+    
+        try {
+            await updateUser(
+                req.session.user._id,
+                req.body.email,
+                req.body.name,
+                req.body.pfp
+            )
+            return res.json({ message: 'User updated.' });
```

Comment:
I would probably like to return the updated user here, just do a getUserById inside of the updateUser function probably would be best